dlt/sources/rest_api/__init__.py

```diff
@@ -410,6 +410,16 @@ def _validate_config(config: RESTAPIConfig) -> None:
         auth = client_config.get("auth")
         if auth:
             auth = _mask_secrets(auth)
+    resources = c.get("resources", [])
+    for resource in resources:
+        if isinstance(resource, str) or isinstance(resource, DltResource):
```

Comment:
I think this can be simplified to:
```suggestion
        if isinstance(resource, (str, DltResource)):
```